File: src/my_py_pkg/my_py_pkg/led_panal.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from the_robot_msgs.msg import LedStateM
from the_robot_msgs.srv import LedState


class LED_Panal(Node): #create the class and inharet from node
    def __init__(self): 
        super().__init__("LED_Panal") #call the init from the Node class 
        self.Leds_server_=self.create_service(LedState,"set_led",self.callback_led_service)
        self.led_state_publisher_=self.create_publisher(LedStateM,"led_state",10)
        self.timer_=self.create_timer(0.5,self.callback_publisher)
        self.responce_= LedState.Response()
        self.get_logger().info("The led server is running....")


    def callback_led_service(self, resquest:LedState.Request ,responce:LedState.Response ):
        if resquest.request[0] == 0 and resquest.request[1] == 0 and resquest.request[2] == 1:
            responce.response[2]= True
            self.responce_=responce
        else:
            responce.response[2]=False
            self.responce_=responce
        return responce
    
    
    
    def callback_publisher(self):
        msg= LedStateM()
        if self.responce_.response[2]==True:
            msg.led_state[2]=True
            self.led_state_publisher_.publish(msg)
        else:
            msg.led_state[2]=False
            self.led_state_publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)   #init the communication
    node=LED_Panal()   #create a object from the class
    rclpy.spin(node)        #make the node spin
    rclpy.shutdown()        #shutdow the node



#if the name main called in the terminal call the main function
if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/led_Panal.cpp
#include "rclcpp/rclcpp.hpp" //importing the ros2 lib if it shows a error open vs from the terminal
#include "the_robot_msgs/srv/led_state.hpp"
#include "the_robot_msgs/msg/led_state_m.hpp"
using namespace std::placeholders;
class LedPanal : public rclcpp::Node //create a class and inharet from the rclcpp
{
    public: //puplic
        LedPanal() : Node("Led_panal") 
        {
            //assign the v and the methods
            led_panal_publisher_=this->create_publisher<the_robot_msgs::msg::LedStateM>("led_state",10);
            led_panal_server_=this->create_service<the_robot_msgs::srv::LedState>("set_led",
            std::bind(&LedPanal::callback_led_srv,this,_1,_2));
            RCLCPP_INFO(get_logger(),"The_led_panal_cpp_is_running...");

        }   
    private:
    //define the v of the class and the private methods
        rclcpp::Service<the_robot_msgs::srv::LedState>::SharedPtr led_panal_server_;
        rclcpp::Publisher<the_robot_msgs::msg::LedStateM>::SharedPtr led_panal_publisher_;


        void callback_led_srv(const the_robot_msgs::srv::LedState::Request::SharedPtr request , the_robot_msgs::srv::LedState::Response::SharedPtr responce)
        {
            auto msg = the_robot_msgs::msg::LedStateM();
            if (request->request[2]== 1 )
            {
                responce->response[2]=true;
                msg.led_state[2]=true;
                led_panal_publisher_->publish(msg);
            }
            else
            {

                responce->response[2]=false;
                msg.led_state[2]=false;
                led_panal_publisher_->publish(msg);
            }
            RCLCPP_INFO(this->get_logger(),"Rq is executed");
        }
};

int main(int argc,char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<LedPanal>(); //create the shared pointer of the my node
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}
